tests/test_basic.py

```diff
@@ -437,3 +440,37 @@ def needsUnittestUser(self):
         """Skips the test unless it is run as the ``unittest`` user."""
         if not testonly.is_unittest_user():
             self.skipTest("May fail with non-unittest users.")
+
+
+@pytest.fixture(scope="function")
+def with_metadata_checker(monkeypatch):
+    """Intercept requests to arrays and groups list endpoints and checks the URLs."""
+
+    def fake_and_check_request(self, method, url, *args, **kwargs):
+        assert "with_metadata=True" in url
```

Comment:
I honestly don't know which side, client or server, has the responsibility for getting the values of these boolean options right.

Reply:
I usually opt for the server to be more flexible but we should also standardize it in the openapi definition